### PR TITLE
feat: thorchain lp fees

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- handles proper fee calculation at input, status, and confirm steps
- handles proper wallet balance and fee asset balance, both for sym and asym pool types, with proper messaging to reflect error states

```
TODO
- [x] split asset and fees asset balance detection and reflect insufficient balance at view-layer
- [x] handle RUNE fees detection as well and also surface at view-layer
- [x] ensure this works with all 3 a/sym types
- [x] surface gas fees at view-layer as well
  - [x] asset
  - [x] RUNE
- [x] confirm step
- [x] status step
```
## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to  https://github.com/shapeshift/web/issues/5960

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

NOTES:

1. Test extensively with native - **do not test with MM** as it will give false positives (or rather negatives) while wallet feature detection isn't implemented yet
2. Keep in mind DOGE trading is currently paused and we do not handle proper halted states yet - do not test UTXOs with DOGE here

- Ensure not enough native balance for gas on any or both of the assets triggers an error / disabled state
- Ensure not enough balance for asset (either native or token) on any or both of the assets triggers an error / disabled state
- Ensure gas fee is wired up at input, confirm, and status steps

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

#### Asset as a pool asset - asym asset side

https://github.com/shapeshift/web/assets/17035424/50364089-d750-4bfa-b274-51b0677760e3

#### Asym RUNE side

https://github.com/shapeshift/web/assets/17035424/82764685-6dbf-4c67-b180-857994063aca

#### Sym

https://github.com/shapeshift/web/assets/17035424/8e7e2d98-f1d1-48a6-af76-15f7264031e5

### Fee balance checks -  EVM token

https://github.com/shapeshift/web/assets/17035424/e08b163e-74c0-4b52-8f84-10bba9ee6817

### Fee balance checks -  ATOM

https://github.com/shapeshift/web/assets/17035424/2d0bcbb8-eaa5-4b63-9558-905277ceb894

### Fee balance checks -  UTXO

https://github.com/shapeshift/web/assets/17035424/fbae6ba5-5fb6-4822-ade9-8d51bf0eec9d

### Confirm step fees - EVM token

image

### Confirm step fees - EVM native asset

image

### Confirm step fees - ROON

image

### Confirm step fees - UTXO

image

### Confirm step fees - ATOM

image

### Status step fees - EVM native asset

image

### Status step fees - ATOM

image

### Status step fees - UTXO

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6127?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #6126** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6126?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6127** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6127?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6127?utm_source=stack-comment).

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -589,15 +617,77 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     selectPortfolioCryptoBalanceBaseUnitByFilter(state, runeBalanceFilter),
   )
 
+  // We reuse lending utils here since all this does is estimating fees for a given deposit amount with a memo
+  // It's not going to be 100% accurate for EVM chains as it doesn't calculate the cost of depositWithExpiry, but rather a simple send,
+  // however that's fine for now until accurate fees estimation is implemented
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:622]: Same things as previous calc - except for RUNE

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 2]
```diff
@@ -230,6 +241,127 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
...
+  const estimateFeesArgs = useMemo(() => {
+    if (!assetId || !wallet || !asset || !poolAsset) return undefined
+
+    const thorchainNotationAssetId = assetIdToPoolAssetId({
+      assetId: isRuneTx ? poolAsset.assetId : asset.assetId,
+    })
+    const transactionType = getThorchainLpTransactionType(asset.chainId)
+    const amountCryptoBaseUnit = toBaseUnit(amountCryptoPrecision, asset.precision)
+
+    switch (transactionType) {
```
Comment (0xApotheosis) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:253]: Can we use `estimateFeesArgs` instead of duplicating this logic?

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Functional testing:

❌ I'm getting "Not enough RUNE to cover gas", and no gas fee calculated for a RUNE-side asym deposit of DAI/RUNE LP.

Screenshot 2024-02-07 at 9 58 28 am

I'm getting the correct error, however, of "Not enough ETH to cover gas" in the generic LP deposit component.

Screenshot 2024-02-07 at 10 04 37 am

Nonblocking, we'll want to unify the precision between the quote and confirm screens:

Screenshot 2024-02-07 at 9 52 10 am
Screenshot 2024-02-07 at 9 52 17 am

~Possible loading state regression here with slippage calculation:~ nope nevermind, this is already in `develop`:

https://github.com/shapeshift/web/assets/97164662/e16f2715-60bf-42a2-b303-44862942d684

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:253]: Discussed this one out, we can't do this unfortunately as `estimateFees()` expects a conssitent args shape, but the various Tx building utils don't

**Review by 0xApotheosis — APPROVED** [review]
All feedback addressed, great PR ser

**Pull request merged**